protocol/xmrmaker/message_handler.go

```diff
@@ -72,7 +72,7 @@ func (s *swapState) setNextExpectedEvent(event EventType) error {
 		panic("status corresponding to event cannot be UnknownStatus")
 	}
 
-	s.info.SetStatus(status)
+	s.UpdateStatus(status)
 	err := s.Backend.SwapManager().WriteSwapToDB(s.info)
```

Comment:
When do we write the the DB after updating the status and when do we not do it?

Reply:
@noot Did you see this question? It's not critical to the PR. I just wanted to understand it better.